aten/src/ATen/native/Linear.cpp

```diff
@@ -0,0 +1,34 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+
+namespace at { namespace native {
+
+Tensor bilinear(const Tensor& input1, const Tensor& input2, const Tensor& weight, const Tensor& bias) {
+  if (input1.dim() != input2.dim()) {
+    throw std::runtime_error("Inputs should have the same number of dimensions");
```

Comment:
Nit: these should be more descriptive in general. "Inputs should have same number of dimensions: got %d and %d".